animalcontrol.lua

```diff
@@ -0,0 +1,184 @@
+-- animalcontrol helps manage the butchery and gelding of animals
+-- Written by Josh Cooper(cppcooper) on 2020-02-18, last modified: 2020-02-19
+local utils=require('utils')
+local validArgs = utils.invert({
+ 'all',
+ 'id',
+ 'race',
+ 'markedfor',
+ 'notmarkedfor',
+ 'gelded',
+ 'notgelded',
+ 'male',
+ 'female',
+
+ 'showstats',
+ 'markfor',
+ 'unmarkfor',
+
+ 'help',
+})
+local args = utils.processArgs({...}, validArgs)
+local help = [====[
+
+animalcontrol
+=============
+Animal control is a script useful for deciding what animals to butcher and geld.
+
+I developed this because I found Dwarf Therapist had a bug for my version of
+dwarf fortress which prevented the geld status of animals from updating. So
+I would be queuing animals to be gelded that had already been gelded.
+
+While not as powerful as Dwarf Therapist in managing animals - in so far as
+DT allows you to sort by various stats and flags - this script does provide
+many options for filtering animals. Additionally you can mark animals for
+slaughter or gelding, you can even do so enmasse if you so choose.
+
+Examples:
+  [DFHack]# animalcontrol -race 170
+  [DFHack]# animalcontrol -race 170 -male -notgelded -showstats
+  [DFHack]# animalcontrol -markfor gelding -id 1988
+  [DFHack]# animalcontrol -markfor slaughter -id 1988
+  [DFHack]# animalcontrol -gelded -markedfor slaughter -unmarkfor slaughter
```

Comment:
Oh, I missed the "definition list" part of the error - Sphinx might be complaining about these examples, the list below not having a blank line before it, or something else (I don't know if the line number is relative to the start of the docs or the file). I would advise pasting these docs into an online RST editor, like http://rst.ninjs.org/, and editing until it's valid.

Reply:
This is still an issue - the docs look like this when rendered (below). I would suggest looking at other scripts for examples of how they format their docs, such as:
* [add-recipe](https://github.com/DFHack/scripts/blob/master/add-recipe.lua) ([rendered](https://docs.dfhack.org/en/latest/docs/_auto/base.html#add-recipe))
* [modtools/create-item](https://github.com/DFHack/scripts/blob/master/modtools/create-item.lua) ([rendered](https://docs.dfhack.org/en/latest/docs/_auto/modtools.html#modtools-create-item)) (this one is just a preformatted block of text)
* [modtools/pref-edit](https://github.com/DFHack/scripts/blob/master/modtools/pref-edit.lua) ([rendered](https://docs.dfhack.org/en/latest/docs/_auto/modtools.html#modtools-pref-edit))
* [modtools/pref-skill-change](https://github.com/DFHack/scripts/blob/master/modtools/skill-change.lua) ([rendered](https://docs.dfhack.org/en/latest/docs/_auto/modtools.html#modtools-skill-change))


For reference, I'm building the docs with `docs/build.sh` in the main DFHack repo, but we have a [new page](https://docs.dfhack.org/en/latest/docs/Documentation.html) that goes into some more detail about building the docs. 

![image](https://user-images.githubusercontent.com/3719547/86491914-3d27e900-bd3a-11ea-890e-85ad31d03dae.png)

Reply:
I fixed this in 988d7d1 and was able to keep pretty much the same formatting in the source code. It looks like this now (rendered as HTML):

![image](https://user-images.githubusercontent.com/3719547/88997792-8d08ba00-d2be-11ea-94d1-4483f5468524.png)